zp-core/zp-extensions/downloadList.php

```diff
@@ -676,8 +711,9 @@ function printDownloadAlbumZipURL($linktext = NULL, $albumobj = NULL, $fromcache
 			$fromcache = getOption('downloadList_zipFromCache');
 		}
 		$success = AlbumZip::create($item, $fromcache);
-		if($success) {
-			exitZP();
+		if(!$success) {
+			$_zp_downloadfile = $item . '.zip';
+			DownloadList::noFile();
```

Comment:
Actually here if "$success=true" nothing happens, because the lib-zipStream.php library has correctly ended its work, hence terminating the execution of the script. Instead, if an album is existing but empty, or the "From Cache" option of the plugin is enabled but the cache is empty, an error message with exit zenphoto is needed.
Perhaps a notice about empty cache should be added to the option description. Something like:
"Only files actually existing in cache at the moment of requesting the download will be added to the zip file".